src/dbsync/utils/pipelinePattern.h

```diff
@@ -0,0 +1,82 @@
+#ifndef PIPELINE_H
+#define PIPELINE_H
+#include <memory>
+#include <vector>
+#include <functional>
+#include "threadDispatcher.h"
+
+namespace Utils
+{
+	// /**
+	//  * @brief Pipeline Reader policy class minimun interface
+	//  * @details Receives messages from a Pipeline Writer class
+	//  * 
+	//  * @tparam Type Message type to be received from a writer.
+	//  */
+	// template<typename Type>
+	// class PipeLineReader
+	// {
+	// public:
+	// 	/**
+	// 	 * @brief Receives a message from a writer to be processed.
+	// 	 * 
+	// 	 * @param data Message data
+	// 	 */
+	// 	void receive(const Type& data);
+	// };
+
+	/**
+	 * @brief Pipeline Writer interface
+	 * @details Base class for Write nodes in the pipeline.
+	 * 
+	 * @tparam T Message type to be sent to readers in the pipeline.
+	 * @tparam Reader Type of the readers that will be chained with the writer.
+	 */
+	template<typename T, typename Reader>
+	class IPipelineWriter
+	{
+	protected:
+		/**
+		 * @brief Sends a message to all the chained readers in the pipeline.
+		 * 
+		 * @param data Message data to be sent to readers.
+		 */
+		void send(const T& data)
+		{
+			for (auto& reader : m_readers)
```

Comment:
could you change this to const ?

ex:
for (const auto& reader : m_readers)

Reply:
will do.